lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/YarnContainerManager.java

```diff
@@ -535,13 +537,28 @@ private void handleNewContainer(final Container container) {
       LOG.log(Level.FINEST, "Allocated Container: memory = {0}, core number = {1}",
           new Object[] {container.getResource().getMemory(), container.getResource().getVirtualCores()});
 
+      YarnClient client = YarnClient.createYarnClient();
+      client.init(new YarnConfiguration());
+      client.start();
+
+      Set<String> nodeLabelExpressions = null;
+
+      try {
+        nodeLabelExpressions = client.getNodeToLabels().get(container.getNodeId());
+      } catch (YarnException e) {
+        e.printStackTrace();
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
```

Comment:
don't handle exceptions like that. instead, write something like:
```java
final Set<String> nodeLabelExpressions;
try {
  nodeLabelExpressions = client.getNodeToLabels().get(container.getNodeId());
} catch (final YarnException | IOException e) {
  LOG.log(Level.WARN, "Error getting labels for node: " + container.getNodeId(), e);
  nodeLabelExpressions = null; // or a new empty Set?
}
```
note:
   * `final` variable `nodeLabelExpressions`
   * `final` exception
   * use of `|` in catch
   * use of `Logger.log(Level, String, Throwable)` specialization instead of `.printStackTrace()`